groups/views.py

```diff
@@ -1,3 +1,161 @@
+    _search = False
+    _search_label = 'Search'
+    paginate_by = PAGINATOR
+    permission_required = 'groups.add_group'
+
+    def form_invalid(self, form):
+        """Overrides parent function to take object_list if form is invalid"""
+        self.object_list = self.get_queryset()
+        return super().form_invalid(form=form)
+
+    def get_context_data(self, **kwargs):
+        context = super(CreateGroupView, self).get_context_data(**kwargs)
+        context['search_label'] = self._search_label
+        context['search'] = self._search
+        return context
+
+    def get_queryset(self):
+        """List of mentors. Search by last_name is available"""
+        result = get_user_model().objects. \
+            filter(is_active=True, groups__name='Mentor').order_by('last_name')
+        if self.request.GET.get('data_search'):
+            search = self.request.GET.get('data_search')
+            result = result.filter(last_name__icontains=search)
+            self._search = True
+            self._search_label = search
+        return result
+
+
+class UpdateGroupView(PermissionRequiredMixin, SuccessMessageMixin,
+                      SafePaginationMixin, generic.UpdateView,
+                      generic.ListView):
+    """
+    Updates group of students.
+    Allows to change name of group and to add or remove mentors
+    """
+    form_class = GroupCreateForm
+    template_name = 'groups/group_update.html'
+    success_message = "Group was updated successfully"
+    _search = False
+    _search_label = 'Search'
+    _checked_mentors = None
+    paginate_by = PAGINATOR
+    permission_required = 'groups.change_group'
+
+    def form_invalid(self, form):
+        """Overrides parent function to take object_list if form is invalid"""
+        self.object_list = self.get_queryset()
+        return super().form_invalid(form=form)
+
+    def get_context_data(self, **kwargs):
+        context = super(UpdateGroupView, self).get_context_data(**kwargs)
+        context['search_label'] = self._search_label
+        context['checked_mentors'] = self._checked_mentors
+        context['search'] = self._search
+        return context
+
+    def get_object(self, context=None):
+        """Returns Group that we are working with"""
+        context = get_object_or_404(Group, pk=self.kwargs['pk'])
+        return context
+
+    def get_queryset(self):
+        """
+        Remembers all checked_mentors of group and then
+        concatenates them with unchecked mentors, so that
+        checked mentors are always first in list
+        Search by last_name of mentor is available
+        """
+        checked_mentors = get_user_model().objects.filter(
+            is_active=True, mentor_in_group=self.kwargs['pk']). \
+            order_by('last_name')
+        self._checked_mentors = checked_mentors
+        mentors = get_user_model().objects. \
+            filter(is_active=True, groups__name='Mentor'). \
```

Comment:
Can you remove magic string 'Mentor'? You can use for example METOR_ID from CultureAnalyzer.constants

Reply:
changed to groups__pk=MENTOR_ID (imported from CultureAnalyser/constants)